src/Api/index.js

```diff
@@ -1062,76 +1063,46 @@ function ToastifyDebug(props) {
-      fetcher: fetcher(undefined, LentMyc),
+      fetcher: fetcher(undefined, RewardsTracker, mycTokenAddress)
     }
   );
 
-  const { data: pendingMycDepositsInStaking } = useSWR(
-    cycle
-      ? [`useStakingApr:pendingMycInStaking:${ARBITRUM}`, ARBITRUM, getContract(ARBITRUM, "LentMYC"), "pendingDeposits"]
-      : null,
+  const esMycTokenAddress = getContract(ARBITRUM, 'ES_MYC');
+  const { data: esMycDeposited } = useSWR(
+    [`useStakingApr:totalDepositSupply(esMYC):${ARBITRUM}`, ARBITRUM, getContract(ARBITRUM, "MYCStakingRewards"), "totalDepositSupply"],
     {
-      fetcher: fetcher(undefined, LentMyc),
+      fetcher: fetcher(undefined, RewardsTracker, esMycTokenAddress)
     }
   );
 
-  const { data: prev } = useSWR(
-    cycle
-      ? [`useStakingApr:prev:${ARBITRUM}`, ARBITRUM, getContract(ARBITRUM, "LentMYC"), "cycleCumulativeEthRewards"]
-      : null,
-    {
-      fetcher: fetcher(undefined, LentMyc, [cycle - 2]),
-    }
-  );
+  useEffect(() => {
+    if(ethPrice?.gt(0) && mycPrice?.gt(0) && tokensPerInterval && mycDeposited && esMycDeposited) {
 
-  const { data: current } = useSWR(
-    cycle
-      ? [`useStakingApr:current:${ARBITRUM}`, ARBITRUM, getContract(ARBITRUM, "LentMYC"), "cycleCumulativeEthRewards"]
-      : null,
-    {
-      fetcher: fetcher(undefined, LentMyc, [cycle - 1]),
-    }
-  );
+      const tokensPerYear = tokensPerInterval.mul(SECONDS_PER_YEAR);
+      const annualRewardsUsd = tokensPerYear.mul(ethPrice);
 
-  const { data: cycleAssets } = useSWR(
-    cycle
-      ? [`useStakingApr:cycleAssets:${ARBITRUM}`, ARBITRUM, getContract(ARBITRUM, "LentMYC"), "cycleSharesAndAssets"]
-      : null,
-    {
-      fetcher: fetcher(undefined, LentMyc, [cycle - 1]),
-    }
-  );
+      const totalDepositTokens = mycDeposited.add(esMycDeposited);
+      const totalDepositsUsd = totalDepositTokens.mul(mycPrice);
 
-  useEffect(() => {
-    const values = [mycAssetsInStaking, pendingMycDepositsInStaking, ethPrice, mycPrice, current, prev, cycleAssets];
-    if (values.every(Boolean)) {
-      const cycleEthRewardsPerShare = current.sub(prev);
-      const cycleSupply = cycleAssets[0];
-      const cycleEthRewards = cycleEthRewardsPerShare.mul(cycleSupply).div(ethers.BigNumber.from(10).pow(18));
-      const ethDistributed = cycleEthRewards;
-      const mycDeposited = mycAssetsInStaking.add(pendingMycDepositsInStaking).div(expandDecimals(1, ETH_DECIMALS));
-      const mycUSDValue = mycDeposited.mul(mycPrice);
-      const ethUSDValue = ethDistributed.mul(ethPrice);
+      const aprPrecision = 10;
+      const apr = annualRewardsUsd.mul(expandDecimals(1, aprPrecision)).div(totalDepositsUsd);
 
-      const aprPercentageForCycle = ethers.utils.formatUnits(ethUSDValue.div(mycUSDValue));
-      const aprPercentageYearly = parseFloat(aprPercentageForCycle) * FORTNIGHTS_IN_YEAR * 100;
-      setStakingApr(aprPercentageYearly.toFixed(2));
+      const formattedApr = apr.toNumber() / (10 ** aprPrecision) * 100;
```

Comment:
did you use numbers here so you got the decimals?

Reply:
yeah pretty much, the precision I chose was arbitrary but should be plenty

Reply:
yeah was just thinking tou could keep in bignumbers until the toFixed but think this is chill